include/oneapi/dpl/pstl/algorithm_fwd.h

```diff
@@ -798,6 +798,23 @@ __pattern_sort(_ExecutionPolicy&&, _RandomAccessIterator, _RandomAccessIterator,
                /*is_parallel=*/::std::true_type,
                /*is_move_constructible=*/::std::true_type);
 
+//------------------------------------------------------------------------
+// sort_by_key
+//------------------------------------------------------------------------
+template <typename _ExecutionPolicy, typename _RandomAccessIterator1, typename _RandomAccessIterator2,
+          typename _Compare, typename _IsVector>
+oneapi::dpl::__internal::__enable_if_host_execution_policy<_ExecutionPolicy, void>
+__pattern_sort_by_key(_ExecutionPolicy&& __exec, _RandomAccessIterator1 __keys_first,
+                      _RandomAccessIterator1 __keys_last, _RandomAccessIterator2 __values_first, _Compare __comp,
+                      _IsVector /*vector=*/, /*is_parallel=*/::std::false_type) noexcept;
+                      
+template <typename _ExecutionPolicy, typename _RandomAccessIterator1, typename _RandomAccessIterator2,
+          typename _Compare, typename _IsVector>
+oneapi::dpl::__internal::__enable_if_host_execution_policy<_ExecutionPolicy, void>
+__pattern_sort_by_key(_ExecutionPolicy&& __exec, _RandomAccessIterator1 __keys_first,
+                      _RandomAccessIterator1 __keys_last, _RandomAccessIterator2 __values_first, _Compare __comp,
+                      _IsVector /*vector=*/, /*is_parallel=*/::std::true_type);
```

Comment:
One of declarations is not `noexcept`. Is that intentional?

Reply:
Yes, that is  intentional. The all parallel pattern have no `noexcept`, because ones may throw an exception for a host backend at least (bad_alloc).